# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.
# print(data.head())
print(data['children'].unique())

[1 0 3 2 4 5]


Видим, что количество детей у Заемщиков находится в диапазоне от 0 до 5. **Аномальных значений нет**

In [23]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21331 entries, 0 to 21524
Data columns (total 14 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   children               21331 non-null  int64  
 1   days_employed          21331 non-null  float64
 2   dob_years              21331 non-null  int64  
 3   education              21331 non-null  object 
 4   education_id           21331 non-null  int64  
 5   family_status          21331 non-null  object 
 6   family_status_id       21331 non-null  int64  
 7   gender                 21331 non-null  object 
 8   income_type            21331 non-null  object 
 9   debt                   21331 non-null  int64  
 10  total_income           21331 non-null  int64  
 11  purpose                21331 non-null  object 
 12  total_income_category  21331 non-null  object 
 13  purpose_category       21331 non-null  object 
dtypes: float64(1), int64(6), object(7)
memory usage: 2.4+ 

Проверили, что нет данных типа NaN, None ... 
И заметили, что *days_employed* имеет вещественный тип. Поэтому сразу **преобразуем *days_employed* к типу int**:

In [24]:
data['days_employed'] = data['days_employed'].astype(int)
data['children'].sort_values().value_counts()

0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: children, dtype: int64

Изучаем распределение по количеству семей, имеющих детей в дипазоне от 0 до 5.
Видим, что примерно 2/3 заёмщиков - бездетные. Количество семей с одним ребёнком превышает даже сумму семей с 2 - 5 детьми.

Обращаем внимание на то, что у нас только 6 точек (0 - 5 детей), которые мы должны сравнить со значением **возврат кредита в срок**
Разбивать их по категориям, например так:
* бездетные
* с 1 ребёнком
* 2-3 ребёнка 
* многодетные (4+ ребенка)

или так:
* бездетные
* с детьми (1-2 ребенка)
* многодетные (3+ ребенка)
не представляется интресным, так как у нас количество точек для анализа сократится с 6 до 3 - 4.+

**Решаем делать анализ без категоризации, используя весь диапазон (0-5 детей)**

Посмотрим количество заемщиков по категориям с разным количеством детей и количество задолженностей у них:

In [25]:
# Создаем сводную таблицу методом pivot_table()
debt_with_children = data.pivot_table(index='children', values='debt', aggfunc=['count', 'sum'])
debt_with_children

count   sum
           debt  debt
children             
0         14091  1063
1          4808   444
2          2052   194
3           330    27
4            41     4
5             9     0

Дополним таблицу колонкой с процентным соотношением количества задолженностей к количеству заёмщиков по каждой категории семей:

In [26]:
def procent_debt(value):
    return f'{value.sum() / value.count():.2%}'

debt_with_children = data.pivot_table(index='children', values='debt', \
                                      aggfunc=['count', 'sum', procent_debt])
debt_with_children

count   sum procent_debt
           debt  debt         debt
children                          
0         14091  1063        7.54%
1          4808   444        9.23%
2          2052   194        9.45%
3           330    27        8.18%
4            41     4        9.76%
5             9     0        0.00%

**Выводы:** 

1. У бездетных семей задолженность реже встречается, чем у семей с детьми.
2. Нулевое значение для семей с 5 детьми рассматривать не будем, так как для них и выборка мала - всего 9 случаев!
3. При довольно ровных результатах для семей с 1,2 и 4 детьми наблюдается резкое снижение просрочки займа для семей с 3 ребенками :-) ! Скорее всего это объясняется господдержкой многосемейных и целевыми выплатами на 3 ребенка.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Изучим категории граждан по семейному положению:

In [27]:
print(data['family_status'].unique())

['женат / замужем' 'гражданский брак' 'вдовец / вдова' 'в разводе'
 'Не женат / не замужем']


Посмотрим зависимость наличия задолженности от семейного положения. Создадим сводную таблицу:

In [28]:
debt_with_family_status = data.pivot_table(index='family_status', values='debt', \
                                      aggfunc=['count', 'sum', procent_debt])
debt_with_family_status

count  sum procent_debt
                        debt debt         debt
family_status                                 
Не женат / не замужем   2796  273        9.76%
в разводе               1189   84        7.06%
вдовец / вдова           951   63        6.62%
гражданский брак        4134  385        9.31%
женат / замужем        12261  927        7.56%

Отсотрируем количество просроченных займов по относительному их количеству:

In [29]:
debt_with_family_status = debt_with_family_status.sort_values(by=('procent_debt', 'debt'))
debt_with_family_status

count  sum procent_debt
                        debt debt         debt
family_status                                 
вдовец / вдова           951   63        6.62%
в разводе               1189   84        7.06%
женат / замужем        12261  927        7.56%
гражданский брак        4134  385        9.31%
Не женат / не замужем   2796  273        9.76%

**Выводы:** 

1. По вероятности просрочки займов можно выделить 3 группы граждан:
    * люди потерявщие супруга в силу каких-то трагических обстоятельств,
    * люди, живущике в браке или находящиеся в разводе,
    * люди, не имеющие официальных семейных обязательств.
    
    
2. Самыми надежными заемщиками являются люди, пережившие трагедию с потерей супруга. Кроме более высокой финансовой дисциплины, возможно, расплачиваться по кредитам им помогают:
    * совместные накопления, 
    * пенсии на потерю супруга, 
    * наследство, 
    * ещё какие-то выплаты...


3. Люди, не имеющие семейных обязательств, относятся к группе с наибольшим риском по образованию финансовой задолжености.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Ранее было установлено: **10  total_income           21331 non-null  int64** 

Таким образом, в значениях ежемесячного дохода нет пропусков и нет значений, кроме целочисленных:

In [30]:
print('Диапазон зарплат Клиентов:', data['total_income'].unique())
print('Минимальное значение зарплаты заёмщиклов:', \
      '{:,}'.format(data['total_income'].min()).replace(',', ' '))
print('Максимальное значение зарплаты заёмщиклов:', \
      '{:,}'.format(data['total_income'].max()).replace(',', ' '))

Диапазон зарплат Клиентов: [253875 112080 145885 ...  89672 244093  82047]
Минимальное значение зарплаты заёмщиклов: 20 667
Максимальное значение зарплаты заёмщиклов: 2 265 604


Введем **категоризацию** по ежемесячному доходу (руб) граждан с поддиапазонами:

* '**F**' - 0 ... 50 000
* '**E**' - 50 001  ... 100 000
* '**D**' - 100 001 ... 200 000
* '**C**' - 200 001 ... 500 000
* '**B**' - 500 001 ... 1 000 000
* '**A**' - 1 000 001 и более
    

In [31]:
def categorize_salary(salary):
    if 0 <= salary <= 50000:
        return 'F'
    elif 50001 <= salary <= 100000:
        return 'E'
    elif 100001 <= salary <= 200000:
        return 'D'
    elif 200001 <= salary <= 500000:
        return 'C'
    elif 500001 <= salary <= 1000000:
        return 'B'
    elif salary >= 1000001:
        return 'A'

print(categorize_salary(55000))
print(categorize_salary(550000))

E
B


In [32]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21331 entries, 0 to 21524
Data columns (total 14 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   children               21331 non-null  int64 
 1   days_employed          21331 non-null  int64 
 2   dob_years              21331 non-null  int64 
 3   education              21331 non-null  object
 4   education_id           21331 non-null  int64 
 5   family_status          21331 non-null  object
 6   family_status_id       21331 non-null  int64 
 7   gender                 21331 non-null  object
 8   income_type            21331 non-null  object
 9   debt                   21331 non-null  int64 
 10  total_income           21331 non-null  int64 
 11  purpose                21331 non-null  object
 12  total_income_category  21331 non-null  object
 13  purpose_category       21331 non-null  object
dtypes: int64(7), object(7)
memory usage: 2.9+ MB


Не будем создавать новый столбец, а поместим в существующий **total_income_category** значения категорий зарплат заемщиков

In [33]:
data['total_income_category'] = data['total_income'].apply(categorize_salary)
data.head()

children  days_employed  dob_years education  education_id  \
0         1           8437         42    высшее             0   
1         1           4024         36   среднее             1   
2         0           5623         33   среднее             1   
3         3           4124         32   среднее             1   
4         0         340266         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     C  операции с недвижимостью  
1     приобретение автомобиля                     D    операции с автомобилем  
2               покупка жилья                     D  операции с недвижимостью  
3  дополнительное образование                     C     получение образования  
4             сыграть свадьбу                     D        проведение свадьбы

Посмотрим статистику по категориям зарплат:

In [34]:
data['total_income_category'].value_counts()

D    11848
C     4817
E     4073
F      371
B      197
A       25
Name: total_income_category, dtype: int64

In [35]:
debt_with_income = data.pivot_table(index=['total_income_category'], values=['debt'], \
                                    aggfunc=['sum', 'count', procent_debt])
debt_with_income

sum  count procent_debt
                       debt   debt         debt
total_income_category                          
A                         2     25        8.00%
B                        12    197        6.09%
C                       342   4817        7.10%
D                      1022  11848        8.63%
E                       331   4073        8.13%
F                        23    371        6.20%

**Категории зарплат:**

* 'A' - 1 000 001 и более
* 'B' - 500 001 ... 1 000 000
* 'C' - 200 001 ... 500 000
* 'D' - 100 001 ... 200 000
* 'E' - 50 001 ... 100 000
* 'F' - 0 ... 50 000

**Выводы:** 

1. Категорию миллионеров рассматривать не будем в силу ее малочисленности.


2. Всех заемщиков можно объединить в 3 группы

* **Надежные**.
      Сверхбогатые (с доходом выше 500000 рублей в месяц) + бедные (с доходом до 50000 руб/мес).
      Они менее других имеют проблемы с выплатой займов.
      
      
* **Безнадежные**. 
      Люди со средним достатком (50 000 - 500 000 руб/мес). 
      Эти заемщики чаще других люднй набирают долги по погашению кредитов.
      
      
* **Вполне надежные**. 
      Это люди, зарабатывающие 200 - 500 тыс рублей в месяц. По уровню просрочек 
      кредитов они находятся не по середине между предудущими двумя группами, а ближе к надежным.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Воспользуемся ранне созданным столбцом **purpose_category** с информацией по целям кредитования:

In [36]:
data['purpose_category'].unique()

array(['операции с недвижимостью', 'операции с автомобилем',
       'получение образования', 'проведение свадьбы'], dtype=object)

In [37]:
debt_purpose = data.pivot_table(index=['purpose_category'], values=['debt'], \
                                    aggfunc=['sum', 'count', procent_debt])
debt_purpose = debt_purpose.sort_values(by=('procent_debt', 'debt'))
debt_purpose

sum  count procent_debt
                         debt   debt         debt
purpose_category                                 
операции с недвижимостью  780  10751        7.26%
проведение свадьбы        183   2313        7.91%
получение образования     369   3988        9.25%
операции с автомобилем    400   4279        9.35%

**Выводы:** 

1. По целям кредитования можно выделить 2 группы с различными значениями появления задолженностей.

    * **С меньшим риском образования задолженности**. К ней можно отнести заёмщиков для операций с 
        недвижимостью и новобрачных.
    * **С повышенным уровнем получения задолженности**. В такую группу попадают люди, берущие кредиты
        на получение образования и операции с автомобилем.  
        
        
2. Выдача кредита на операции с недвижимостью - наименее рисковая.


3. Удивляет довольно большое (около 8%) количество должников по кредитам на проведение свадьбы. 
    Видимо, люди частенько (у нас 183 раза) женятся на последние деньги (
    

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 
Такие проблемы могут быть вызваны:
* неверным вводом данных, 
* сокрытием информации, 
* повреждением файла данных при сохранении, 
* техническими проблемами.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 

В отличие от кажущимся более привычным **средним значением** медиана менее подвержена влиянию аномальных значений. И если при пользовании **средним** даже одно аномальное число может сильно изменить усредненное значение, то на медианное значение это скажется куда как меньше.

### Шаг 4: общий вывод.

Общие выводы иногда могут вызвать улыбку, но, главное, они не предсказывают аномально опасных типов заёмщиков:

1. Идеальный заёмщик:
   *	Бездетный человек, потерявший супруга,
   *	живущий с доходом свыше полумиллиона в месяц или, наоборот, на пенсию/пособие,
   *	Проводящий операции с недвижимостью (наверное, для богатых людей) или готовящийся к свадьбе (видимо, пенсионер) :-)


2. Самый опасный заёмщик:
   *	одинокий человек, не знавший горечи утраты супруга,
   *	обременённый детьми, 
   *	со средним достатком,
   *	покупающий автомобиль.
   
   
3. Чаще всего кредиты оформляют семейные пары (возможно, могут поручиться друг за друга) с детьми, со средним доходом (100-200 тыс руб) на покупку недвижимости.


4. Реже оформляют кредиты люди, потерявшие супруга, с доходом или до 1 млн руб.мес или до 50 тыс руб.мес. на покупку недвижимости или на новую свадьбу. Практически редчайший случай, когда у них ещё и 5 детей.


5. По различным факторам (уровень дохода, наличие детей, семейное положение и целям кредитования) отличие максимального и минимального вероятности возникновения задолженности по выплате кредита не превышает 20-30% относительных. Таким образом, среди перечисленных факторов нет таких, которые могли бы аномально увеличить риск невозврата кредита.


[Вернутся к общему списку проектов курса “Data Science” Яндекс-практикума](../README.md)